### ISSUE TITLE
SVG Exporter does not place symbols on correct place

### ISSUE THREAD

**Issue by CPrescher** [pyqtgraph/pyqtgraph#176]
Hei there,

i have found a strange bug, when creating a plot having a line and markers for a specific dataset the SVG Exporter does not place the marker symbols at the expected positions but somehow below.

an example script:

```
import pyqtgraph as pg
from pyqtgraph.exporters.SVGExporter import SVGExporter
from PyQt4 import QtGui

if __name__ == '__main__':
    app = QtGui.QApplication([])
    win = pg.GraphicsWindow("title SVG Exporter bug")
    win.resize(1000, 600)
    p1 = win.addPlot()
    p1.plot(x=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10], y=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10], pen=(200, 200, 200),
            symbolBrush=(255, 0, 0), symbolPen='w')
    QtGui.QApplication.processEvents()
    QtGui.QApplication.processEvents()
    exporter = SVGExporter(p1)
    exporter.export("test.svg")``
```

This produces an svg file were the markers are clearly at the wrong positions...

![plotitem_1](https://cloud.githubusercontent.com/assets/4086557/8243952/9269077e-15e1-11e5-859b-3c80fdd4bd67.png)

**Comment by markotoplak** [comment]
The same thing happens with an ImageExporter (to .png).

**Comment by j9ac9k** [comment]
Confirmed that this is still the case with the current developer build.  Thanks for the example code/error.

**Comment by paulmueller** [comment]
I can also confirm this bug.

When explicitly plotting the scatter plot with "pxMode=False", the offset goes away:
```
import pyqtgraph as pg
import numpy as np
from pyqtgraph.exporters.SVGExporter import SVGExporter
from PyQt5 import QtGui

if __name__ == '__main__':
    app = QtGui.QApplication([])
    win = pg.GraphicsWindow("title SVG Exporter bug")
    win.resize(1000, 600)
    p1 = win.addPlot()
    p1.plot(x=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10], y=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
    scatter = pg.ScatterPlotItem(pen=(200, 200, 200),
                                 symbolBrush=(255, 0, 0),
                                 symbolPen='w',
                                 size=.5,
                                 pxMode=False,
                                 )
    p1.addItem(scatter)
    scatter.setData(np.arange(10), np.arange(10))

    QtGui.QApplication.processEvents()
    QtGui.QApplication.processEvents()
    exporter = SVGExporter(p1)
    exporter.export("test2.svg")
```
![rect992](https://user-images.githubusercontent.com/2853022/69483258-4f958800-0e25-11ea-9f8c-e5daf0ce7ca6.png)

Note that:
 - This bug appears only if `pxMode == True`
 - Here, the circles are transformed along with the GraphicsWindow size (setting rectangular GraphicsWindow size makes them circular again). There is something wrong with the transforms performed in ScatterPlotItem.
 - Setting `pxMode=True` and `size=5` reproduces the original bug.

**Comment by paulmueller** [comment]
I have created a separate issue for the scaling problem: #1086

**Comment by paulmueller** [comment]
I believe I have fixed this issue with https://github.com/paulmueller/pyqtgraph/commit/54d3490bbde3d013b56b1ea453bd4a5a9459579c

...but I am hesitant to create a PR, because I really don't know what I did there.

**Comment by paulmueller** [comment]
For the sake of completeness:
- All the changes I made are recorded in https://github.com/paulmueller/pyqtgraph/commits/possible-fix-for-176 (Nov 25tj 2019)
- This only fixes things for PlotItem, not for ViewBox (can be tested by exporting individual plots using right-click export in the ScatterPlot example)

```python
import pyqtgraph as pg
import numpy as np
from pyqtgraph.exporters.SVGExporter import SVGExporter
from pyqtgraph.exporters.ImageExporter import ImageExporter
from PyQt5 import QtGui

if __name__ == '__main__':
    app = QtGui.QApplication([])
    win = pg.GraphicsWindow("PNG export")
    win.resize(1000, 600)
    p1 = win.addPlot()
    p1.plot(x=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10], y=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
    scatter = pg.ScatterPlotItem(pen=(200, 200, 200),
                                 symbolBrush=(255, 0, 0),
                                 symbolPen='w',
                                 size=5,
                                 )
    p1.addItem(scatter)
    scatter.setData(np.arange(10), np.arange(10))

    QtGui.QApplication.processEvents()
    QtGui.QApplication.processEvents()

    pngexp = ImageExporter(p1)
    pngexp.params["width"] = int(pngexp.params["width"])
    pngexp.export("176.png")

    pngexp = ImageExporter(p1)
    pngexp.params["width"] = int(pngexp.params["width"] * 3)
    pngexp.export("176_scaled.png")

    svgexp = SVGExporter(p1)
    svgexp.export("176.svg")
```

176.png:
![176](https://user-images.githubusercontent.com/2853022/69552951-9d360000-0f9f-11ea-8cd8-272c4ddc4a7f.png)

176_scaled.png:
![176_scaled](https://user-images.githubusercontent.com/2853022/69552950-9d360000-0f9f-11ea-82fd-e12ae136a450.png)

176.svg (converted using inkscape):
![176_svg](https://user-images.githubusercontent.com/2853022/69553089-d8383380-0f9f-11ea-9b3e-48784ce3af12.png)

And the screenshot:
![image](https://user-images.githubusercontent.com/2853022/69553168-01f15a80-0fa0-11ea-86ce-f0af009f0e8a.png)

**Comment by j9ac9k** [comment]
Fixed with #1420
